pyaerocom/colocation_auto.py

```diff
@@ -378,6 +390,36 @@ def run(self, var_name=None, **opts):
         finally:
             self._close_log()
 
+    def instantiate_gridded_reader(self, what):
+        """
+        Create reader for model or observational gridded data.
+
+        Parameters
+        ----------
+        what : str
+            Type of reader. ("model" or "obs")
+        """
+        if what == 'model':
+            data_id = self.model_id
+            data_dir = self.model_data_dir
+        else:
+            data_id = self.obs_id
+            data_dir = self.obs_data_dir
+        reader_class = self._get_gridded_reader_class(what=what)
+        reader = reader_class(data_id=data_id, data_dir=data_dir)
+        if hasattr(reader, 'filepath'):
+            reader.filepath = self.filepath
+        return reader
+
+    def _get_gridded_reader_class(self, what):
+        """Returns the class of the reader for gridded data."""
+        try:
+            reader = self.SUPPORTED_GRIDDED_READERS[self.gridded_reader_id[what]]
+        except KeyError as e:
+            raise NotImplementedError('Reader {} is not supported: {}'.
+                                      format(self.gridded_reader_id[what], e))
+        return reader
+
```

Comment:
Perhaps the logic around inferring the correct reader should happen elsewhere and could see some revision / redesign here, but I think this is fine for now and will all be revised when tackling #174.

Reply:
Yes, it should be easier after that.